tests/rptest/tests/describe_topics_test.py

```diff
@@ -167,38 +148,87 @@ def test_describe_topics_with_documentation_and_types(self):
+        # - An empty line
+        # - Three lines for each property, called the documentation section:
+        #     1. Property name with a colon (e.g., cleanup.policy:)
+        #     2. Property documentation string
+        #     3. An empty line
+
+        # First check redpanda.datapolicy in the property table because the value is a special case.
+        # The extra space is necessary so the datapolicy lines in the documentation section are not removed.
+        copro_lines = [
+            line for line in output if line.startswith('redpanda.datapolicy ')
+        ]
+        self.logger.debug(f"Copro lines {copro_lines}")
+        assert len(copro_lines) == 1
+        copro_match = re.match(r"^.*default_config$", copro_lines[0])
+        assert copro_match is not None
+
+        all_other_lines = [
+            line for line in output
+            if not line.startswith('redpanda.datapolicy ')
+        ]
+        property_re = re.compile(
+            r"^(?P<name>[a-z.]+?)\s+(?P<type>[a-z]+?)\s+(?P<value>\S+?)\s+(?P<src>[a-z_]+?)$"
+        )
+
+        last_pos = None
+        for i, line in enumerate(all_other_lines):
+            self.logger.debug(f"Property line {line}")
+            prop_match = property_re.match(line)
+
+            if prop_match is None:
+                last_pos = i
+                break
+
+            name = prop_match.group("name")
+            config_type = prop_match.group("type")
+            value = prop_match.group("value")
+            source_type = prop_match.group("src")
+            self.logger.debug(
+                f"name: {name}, type: {config_type}, value: {value}, src: {source_type}"
+            )
+            assert name in properties
             prop = properties[name]
             assert config_type == prop.config_type
             assert value == prop.value
             assert source_type == prop.source_type
 
-        # The output format follows:
-        # line 1 is empty
-        # line 2 is the property name
-        # line 3 is the property docstring
-        prop = None
-        for i in range(0, len(documentation_lines), 3):
-            # The property names have a colon at the end
-            prop_match = re.match(r"^(?P<name>[A-Za-z.]+?):$", line[1])
-            if prop_match is not None:
-                name = prop_match.group("name").lower()
-                assert name in properties.keys()
-                prop = properties[name]
-                assert line[2].lower() == prop.doc_string
+        # The first empty line is where the table ends and the doc section begins
+        assert last_pos is not None, "Something went wrong with property match"
+        self.logger.debug(
+            f"Table separator {last_pos} {len(all_other_lines[last_pos])}")
+        assert len(all_other_lines[last_pos]) == 0, "Expected empty line"
+
+        # Make a list from the leftover lines
+        doc_lines = all_other_lines[last_pos + 1:]
```

Comment:
nit: no guarantee that `len(all_other_lines) > last_pos + 1` so out of index crash could occur here